services/wallet/collectibles/service.go

```diff
@@ -0,0 +1,259 @@
+		collectibles, hasMore, err := s.manager.GetOwnedCollectibles(chainIDs, addresses, offset, limit)
+		if err != nil {
+			return nil, err
+		}
+		data, err := s.manager.FetchAssetsByCollectibleUniqueID(collectibles)
+		if err != nil {
+			return nil, err
+		}
+
+		return filterOwnedCollectiblesTaskReturnType{
+			collectibles: thirdparty.CollectiblesToHeaders(data),
+			hasMore:      hasMore,
+		}, err
+	}, func(result interface{}, taskType utils.TaskType, err error) {
+		res := FilterOwnedCollectiblesResponse{
+			ErrorCode: ErrorCodeFailed,
+		}
+
+		if errors.Is(err, context.Canceled) || errors.Is(err, utils.ErrTaskOverwritten) {
+			res.ErrorCode = ErrorCodeTaskCanceled
+		} else if err == nil {
+			fnRet := result.(filterOwnedCollectiblesTaskReturnType)
+			res.Collectibles = fnRet.collectibles
+			res.Offset = offset
+			res.HasMore = fnRet.hasMore
+			res.ErrorCode = ErrorCodeSuccess
+		}
+
+		s.sendResponseEvent(EventOwnedCollectiblesFilteringDone, res, err)
+	})
+}
+
+func (s *Service) GetCollectiblesDataAsync(ctx context.Context, uniqueIDs []thirdparty.CollectibleUniqueID) {
+	s.scheduler.Enqueue(getCollectiblesDataTask, func(ctx context.Context) (interface{}, error) {
+		collectibles, err := s.manager.FetchAssetsByCollectibleUniqueID(uniqueIDs)
+		return collectibles, err
+	}, func(result interface{}, taskType utils.TaskType, err error) {
+		res := GetCollectiblesDataResponse{
+			ErrorCode: ErrorCodeFailed,
+		}
+
+		if errors.Is(err, context.Canceled) || errors.Is(err, utils.ErrTaskOverwritten) {
+			res.ErrorCode = ErrorCodeTaskCanceled
+		} else if err == nil {
+			collectibles := result.([]thirdparty.CollectibleData)
+			res.Collectibles = collectibles
+			res.ErrorCode = ErrorCodeSuccess
+		}
+
+		s.sendResponseEvent(EventGetCollectiblesDataDone, res, err)
+	})
+}
+
+func (s *Service) triggerEvent(eventType walletevent.EventType, account statustypes.Address, message string) {
+	s.eventFeed.Send(walletevent.Event{
+		Type: eventType,
+		Accounts: []common.Address{
+			common.Address(account),
+		},
+		Message: message,
+	})
+}
+
+func (s *Service) Start() {
+	go func() {
+		s.serviceContext, s.cancelFn = context.WithCancel(context.Background())
+		s.timer = time.NewTimer(accountOwnershipUpdateInterval)
+
+		update := func() (exit bool) {
+			err := s.updateOwnershipForAllAccounts(s.serviceContext)
+			if s.serviceContext.Err() != nil {
+				s.triggerEvent(EventCollectiblesOwnershipUpdateFinished, statustypes.Address{}, "Service cancelled")
+				s.timer.Stop()
+				return true
+			}
+			if err != nil {
+				s.triggerEvent(EventCollectiblesOwnershipUpdateFinishedWithError, statustypes.Address{}, err.Error())
+			}
+			return false
+		}
+
+		if update() {
+			return
+		}
+
+		for range s.timer.C {
+			s.resetTimer(accountOwnershipUpdateInterval)
+
+			if update() {
+				return
+			}
+		}
+	}()
+}
+
+func (s *Service) Stop() {
+	if s.cancelFn != nil {
+		s.cancelFn()
+	}
+}
```

Comment:
Please stop the scheduler also to cleanup running tasks
See here: https://github.com/status-im/status-go/blob/8926990c2de4aaa2b0a22f7c22aaf5705357d106/services/wallet/activity/service.go#L159